### DISCUSSION TITLE
Programmatically Check pocketbase version

### DISCUSSION THREAD

**Discussion by fbede** [pocketbase/pocketbase#2993] [Q&A] [answered]
Is there a way to find out what version of pocketbase is running from the SDKs? This would be useful to ask users to update their mobile app if the server gets updated and makes the mobile app incompatible

**Comment by ganigeorgiev** [comment, marked as answer]
We don't have web API that output the version at the moment (_it exists in my todo so that we can also make the version in the Admin UI dynamic, but remains a very low priority as there are some complications around the versions when PocketBase is used as framework_).

> This would be useful to ask users to update their mobile app if the server gets updated and makes the mobile app incompatible

Even if there was a version API I'm not sure that I understand how do you plan to implement the check, because your app will need to have upfront information what are the supported PocketBase versions, meaning that with each new PocketBase release you'll have to force your users to update your application even if there isn't a breaking change in the latest PocketBase version.

Ideally once we reach a stable v1 there will be no more breaking changes and you wouldn't have to worry about this, but a better approach maybe could be to have a collection with some sort of _flags_ based on which you'll either have a custom mobile app handling OR some custom code when used as framework that will perform the backward compatibility checks/fallbacks.

**Reply by fbede** [reply]
Actually, I think this is a very simple thing to do. My use case is not too complicated and everything I need has already been built in. Thank you so much for the effort you've put in so far.

As for how to implement the feature I mentioned, it should be simple. Using pseudocode;

```
if(pb.version > 0.16.10){
   askUserToUpdateApp();
}else{
   runApp();
}
```

**Reply by ganigeorgiev** [reply]
But again, that `016.10` number would have to be hardcoded in your app and in the next version of your app you'll have to change it for example to v0.17.0.
The main issue with this approach is that when PocketBase releases a new version your users will be again prompted to update to a new version of your mobile app **even if there are no breaking changes in the PocketBase APIs**, aka. your mobile app would not have upfront information whether the new version is breaking or not.

**Reply by fbede** [reply]
Ohhh... Thank you so much. I now understand what you meant. I think the flags you suggested would be a good use case for the singleton collection type in [#159](https://github.com/pocketbase/pocketbase/issues/159). Thank you for your work so far.

**Comment by DanyWallace** [comment]
You can query the github api with your program and get a json response, I think this is kind of what you want?
You can take a look at "name" or "tagname"

```
https://api.github.com/repos/pocketbase/pocketbase/releases/latest
```
